### PR TITLE
Feature/expose colour in switchfield

### PR DESCRIPTION
**Pull request by jegasriskantha**
contributes to: IPG-Automotive-UK/instrument-designer/issues/121

This pull request: 
- adds new thumbColor and trackColor property to easily change the color properties of the MUI switchField 
- adds new switchStyle property to make it easier to customise the MUI switchField
- conditionally renders the FormHelperText only when the helperText prop is set. 

image

### CODE REVIEW

**Comment by evoinea** [comment]
As discussed, I think it might be a good idea to expose text colours too

File: src/SwitchField/SwitchField.js [hunk 1]
```diff
@@ -105,5 +123,17 @@ FormSwitch.propTypes = {
         `Invalid prop ${propName} supplied to ${componentName}. Expected a string array of length 2.`
       );
     }
-  }
+  },
+  /**
+   * Custom styling for the MUI Switch component.
+   */
+  switchStyle: PropTypes.string,
```
Comment (dmbarry86) [inline src/SwitchField/SwitchField.js:130]: What's the purpose of this switchStyle prop? What would we use it for? If we need something non-standard then we just override the theme in the app where we need to. Also, should it not be an object anyway not a string? I think just remove this.

**Comment by dmbarry86** [comment]
The track color is not actually the color I specify. It's the color I specify with some opacity on top. This won't work how you think it will in ID (which presumably is what you have in mind).

For example here I set both to red but only the thumb is red.
image

To be honest I don't really see why we need to complicate the API with these color props as 99% of the time you would just want it to match the theme of the app, e.g. primary. For the instrument designer usecase we can just write a custom style override (like we do for button color for example).

**Comment by dmbarry86** [comment]
> * conditionally renders the FormHelperText only when the helperText prop is set.

I don't see any code you have added for this? It looks like the code already does this anyway on line 64 which has been there since day 1 of the component. Am I missing something?

**Review by dmbarry86 — COMMENTED** [review]
I'm not convinced about the need for this PR. Let's discuss tomorrow.

**Comment by dmbarry86** [comment]
> As discussed, I think it might be a good idea to expose text colours too

I don't think we should do this. I don't want to start complicating the API of these components with non-standard styling props. If we don't want the default MUi styling in a particular app then we just override it.

**Comment by dmbarry86** [comment]
See https://github.com/IPG-Automotive-UK/instrument-designer/pull/140

**Pull request closed without merging**